src/Chapter4.hs

```diff
@@ -673,6 +713,25 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a = Node a (Tree a) (Tree a) | Leaf deriving (Show)
+
+instance Functor Tree where
+  fmap :: (a -> b) -> Tree a -> Tree b
+  fmap f (Node val l r) = Node (f val) (fmap f l) (fmap f r)
+  fmap _ Leaf = Leaf
+
+reverseTree :: Tree a -> Tree a
+reverseTree (Node val l r) = Node val (reverseTree r) (reverseTree l)
+reverseTree Leaf = Leaf
+
+
+treeToList :: Tree a -> List a
+treeToList (Node val l r) = flatten (list3 (treeToList l) (pure val) (treeToList r))
+  where
+    list3 :: a -> a -> a -> List a
+    list3 a b c = Cons a (Cons b (Cons c Empty))
```

Comment:
Interesting!
I guess you could make it a bit more efficient if you implement append function for List instead. `append` also could be reused for `flatten` implementation.